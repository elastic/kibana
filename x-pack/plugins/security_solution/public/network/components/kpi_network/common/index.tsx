/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import React from 'react';

import { EuiFlexItem, EuiLoadingSpinner, EuiFlexGroup } from '@elastic/eui';
import styled from 'styled-components';

import { manageQuery } from '../../../../common/components/page/manage_query';
import { NetworkKpiStrategyResponse } from '../../../../../common/search_strategy';
import {
  StatItemsComponent,
  StatItemsProps,
  useKpiMatrixStatus,
  StatItems,
} from '../../../../common/components/stat_items';
import { UpdateDateRange } from '../../../../common/components/charts/common';

const kpiWidgetHeight = 228;

export const FlexGroup = styled(EuiFlexGroup)`
  min-height: ${kpiWidgetHeight}px;
`;

FlexGroup.displayName = 'FlexGroup';

export const NetworkKpiBaseComponent = React.memo<{
  fieldsMapping: Readonly<StatItems[]>;
  data: NetworkKpiStrategyResponse;
  loading?: boolean;
  id: string;
  from: string;
  to: string;
  narrowDateRange: UpdateDateRange;
}>(({ fieldsMapping, data, id, loading = false, from, to, narrowDateRange }) => {
  const statItemsProps: StatItemsProps[] = useKpiMatrixStatus(
    fieldsMapping,
    data,
    id,
    from,
    to,
    narrowDateRange
  );

  if (loading) {
    return (
      <FlexGroup justifyContent="center" alignItems="center">
        <EuiFlexItem grow={false}>
          <EuiLoadingSpinner size="xl" />
        </EuiFlexItem>
      </FlexGroup>
    );
  }

  return (
    <EuiFlexGroup wrap>
      {statItemsProps.map((mappedStatItemProps) => (
        <StatItemsComponent {...mappedStatItemProps} />
      ))}
    </EuiFlexGroup>
  );
});

NetworkKpiBaseComponent.displayName = 'NetworkKpiBaseComponent';

export const NetworkKpiBaseComponentManage = manageQuery(NetworkKpiBaseComponent);
