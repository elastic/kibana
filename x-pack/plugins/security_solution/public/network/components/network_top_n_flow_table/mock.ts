/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import {
  NetworkTopNFlowStrategyResponse,
  FlowTargetSourceDest,
} from '../../../../common/search_strategy';

export const mockData: NetworkTopNFlowStrategyResponse = {
  totalCount: 524,
  edges: [
    {
      node: {
        source: {
          autonomous_system: {
            name: 'Google, Inc',
            number: 15169,
          },
          domain: ['test.domain.com'],
          flows: 12345,
          destination_ips: 12,
          ip: '8.8.8.8',
          location: {
            geo: {
              continent_name: ['North America'],
              country_iso_code: ['US'],
              city_name: ['Mountain View'],
              region_iso_code: ['US-CA'],
              region_name: ['California'],
            },
            flowTarget: FlowTargetSourceDest.source,
          },
        },
        network: {
          bytes_in: 3826633497,
          bytes_out: 1083495734,
        },
      },
      cursor: {
        value: '8.8.8.8',
      },
    },
    {
      node: {
        source: {
          autonomous_system: {
            name: 'TM Net, Internet Service Provider',
            number: 4788,
          },
          domain: ['test.domain.net', 'test.old.domain.net'],
          flows: 12345,
          destination_ips: 12,
          ip: '9.9.9.9',
          location: {
            geo: {
              continent_name: ['Asia'],
              country_iso_code: ['MY'],
              city_name: ['Petaling Jaya'],
              region_iso_code: ['MY-10'],
              region_name: ['Selangor'],
            },
            flowTarget: FlowTargetSourceDest.source,
          },
        },
        network: {
          bytes_in: 3826633497,
          bytes_out: 1083495734,
        },
      },
      cursor: {
        value: '9.9.9.9',
      },
    },
  ],
  pageInfo: {
    activePage: 1,
    fakeTotalCount: 50,
    showMorePagesIndicator: true,
  },
  rawResponse: {} as NetworkTopNFlowStrategyResponse['rawResponse'],
};
