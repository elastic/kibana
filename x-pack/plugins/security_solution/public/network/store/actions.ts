/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import actionCreatorFactory from 'typescript-fsa';
import { networkModel } from '.';

const actionCreator = actionCreatorFactory('x-pack/security_solution/local/network');

export const updateNetworkTable = actionCreator<{
  networkType: networkModel.NetworkType;
  tableType: networkModel.NetworkTableType | networkModel.NetworkDetailsTableType;
  updates: networkModel.TableUpdates;
}>('UPDATE_NETWORK_TABLE');

export const setNetworkDetailsTablesActivePageToZero = actionCreator(
  'SET_IP_DETAILS_TABLES_ACTIVE_PAGE_TO_ZERO'
);

export const setNetworkTablesActivePageToZero = actionCreator(
  'SET_NETWORK_TABLES_ACTIVE_PAGE_TO_ZERO'
);
