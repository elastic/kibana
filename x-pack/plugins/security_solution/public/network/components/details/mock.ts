/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

import { NetworkDetailsStrategyResponse } from '../../../../common/search_strategy';

export const mockData: Readonly<
  Record<string, NetworkDetailsStrategyResponse['networkDetails']>
> = {
  complete: {
    source: {
      firstSeen: '2019-02-07T17:19:41.636Z',
      lastSeen: '2019-02-07T17:19:41.636Z',
      autonomousSystem: { organization: { name: 'Test Org' }, number: 12345 },
      geo: {
        continent_name: ['North America'],
        city_name: ['New York'],
        country_iso_code: ['US'],
        country_name: undefined,
        location: {
          lat: [40.7214],
          lon: [-74.0052],
        },
        region_iso_code: ['US-NY'],
        region_name: ['New York'],
      },
    },
    destination: {
      firstSeen: '2019-02-07T17:19:41.648Z',
      lastSeen: '2019-02-07T17:19:41.648Z',
      autonomousSystem: { organization: { name: 'Test Org' }, number: 12345 },
      geo: {
        continent_name: ['North America'],
        city_name: ['New York'],
        country_iso_code: ['US'],
        country_name: undefined,
        location: {
          lat: [40.7214],
          lon: [-74.0052],
        },
        region_iso_code: ['US-NY'],
        region_name: ['New York'],
      },
    },
    host: {
      os: {
        kernel: ['4.14.50-v7+'],
        name: ['Raspbian GNU/Linux'],
        family: [''],
        version: ['9 (stretch)'],
        platform: ['raspbian'],
      },
      name: ['raspberrypi'],
      id: ['b19a781f683541a7a25ee345133aa399'],
      ip: ['10.10.10.10'],
      architecture: ['armv7l'],
    },
  },
};
