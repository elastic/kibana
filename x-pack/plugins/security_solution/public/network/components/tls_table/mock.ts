/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { NetworkTlsStrategyResponse } from '../../../../common/search_strategy';

export const mockTlsData: NetworkTlsStrategyResponse = {
  totalCount: 2,
  edges: [
    {
      node: {
        _id: '2fe3bdf168af35b9e0ce5dc583bab007c40d47de',
        subjects: ['*.elastic.co'],
        ja3: ['7851693188210d3b271aa1713d8c68c2', 'fb4726d465c5f28b84cd6d14cedd13a7'],
        issuers: ['DigiCert SHA2 Secure Server CA'],
        notAfter: ['2021-04-22T12:00:00.000Z'],
      },
      cursor: {
        value: '2fe3bdf168af35b9e0ce5dc583bab007c40d47de',
      },
    },
    {
      node: {
        _id: '61749734b3246f1584029deb4f5276c64da00ada',
        subjects: ['api.snapcraft.io'],
        ja3: ['839868ad711dc55bde0d37a87f14740d'],
        issuers: ['DigiCert SHA2 Secure Server CA'],
        notAfter: ['2019-05-22T12:00:00.000Z'],
      },
      cursor: {
        value: '61749734b3246f1584029deb4f5276c64da00ada',
      },
    },
    {
      node: {
        _id: '6560d3b7dd001c989b85962fa64beb778cdae47a',
        subjects: ['changelogs.ubuntu.com'],
        ja3: ['da12c94da8021bbaf502907ad086e7bc'],
        issuers: ["Let's Encrypt Authority X3"],
        notAfter: ['2019-06-27T01:09:59.000Z'],
      },
      cursor: {
        value: '6560d3b7dd001c989b85962fa64beb778cdae47a',
      },
    },
  ],
  pageInfo: {
    activePage: 1,
    fakeTotalCount: 50,
    showMorePagesIndicator: true,
  },
  rawResponse: {} as NetworkTlsStrategyResponse['rawResponse'],
};
