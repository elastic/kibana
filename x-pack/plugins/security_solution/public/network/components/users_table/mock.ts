/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

import { NetworkUsersStrategyResponse } from '../../../../common/search_strategy';

export const mockUsersData: NetworkUsersStrategyResponse = {
  edges: [
    {
      node: {
        _id: '_apt',
        user: {
          id: ['104'],
          name: '_apt',
          groupId: ['65534'],
          groupName: ['nogroup'],
          count: 10,
        },
      },
      cursor: {
        value: '_apt',
        tiebreaker: null,
      },
    },
    {
      node: {
        _id: 'root',
        user: {
          id: ['0'],
          name: 'root',
          groupId: ['116', '0'],
          groupName: ['Debian-exim', 'root'],
          count: 108,
        },
      },
      cursor: {
        value: 'root',
        tiebreaker: null,
      },
    },
    {
      node: {
        _id: 'systemd-resolve',
        user: {
          id: ['102'],
          name: 'systemd-resolve',
          groupId: [],
          groupName: [],
          count: 4,
        },
      },
      cursor: {
        value: 'systemd-resolve',
        tiebreaker: null,
      },
    },
  ],
  totalCount: 3,
  pageInfo: {
    activePage: 1,
    fakeTotalCount: 3,
    showMorePagesIndicator: true,
  },
  rawResponse: {} as NetworkUsersStrategyResponse['rawResponse'],
};
