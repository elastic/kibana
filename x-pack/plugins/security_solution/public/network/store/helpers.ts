/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import {
  NetworkModel,
  NetworkType,
  NetworkTableType,
  NetworkDetailsTableType,
  NetworkQueries,
  NetworkDetailsQueries,
} from './model';
import { DEFAULT_TABLE_ACTIVE_PAGE } from '../../common/store/constants';

export const setNetworkPageQueriesActivePageToZero = (state: NetworkModel): NetworkQueries => ({
  ...state.page.queries,
  [NetworkTableType.topCountriesSource]: {
    ...state.page.queries[NetworkTableType.topCountriesSource],
    activePage: DEFAULT_TABLE_ACTIVE_PAGE,
  },
  [NetworkTableType.topCountriesDestination]: {
    ...state.page.queries[NetworkTableType.topCountriesDestination],
    activePage: DEFAULT_TABLE_ACTIVE_PAGE,
  },
  [NetworkTableType.topNFlowSource]: {
    ...state.page.queries[NetworkTableType.topNFlowSource],
    activePage: DEFAULT_TABLE_ACTIVE_PAGE,
  },
  [NetworkTableType.topNFlowDestination]: {
    ...state.page.queries[NetworkTableType.topNFlowDestination],
    activePage: DEFAULT_TABLE_ACTIVE_PAGE,
  },
  [NetworkTableType.dns]: {
    ...state.page.queries[NetworkTableType.dns],
    activePage: DEFAULT_TABLE_ACTIVE_PAGE,
  },
  [NetworkTableType.tls]: {
    ...state.page.queries[NetworkTableType.tls],
    activePage: DEFAULT_TABLE_ACTIVE_PAGE,
  },
  [NetworkTableType.http]: {
    ...state.page.queries[NetworkTableType.http],
    activePage: DEFAULT_TABLE_ACTIVE_PAGE,
  },
});

export const setNetworkDetailsQueriesActivePageToZero = (
  state: NetworkModel
): NetworkDetailsQueries => ({
  ...state.details.queries,
  [NetworkDetailsTableType.topCountriesSource]: {
    ...state.details.queries[NetworkDetailsTableType.topCountriesSource],
    activePage: DEFAULT_TABLE_ACTIVE_PAGE,
  },
  [NetworkDetailsTableType.topCountriesDestination]: {
    ...state.details.queries[NetworkDetailsTableType.topCountriesDestination],
    activePage: DEFAULT_TABLE_ACTIVE_PAGE,
  },
  [NetworkDetailsTableType.topNFlowSource]: {
    ...state.details.queries[NetworkDetailsTableType.topNFlowSource],
    activePage: DEFAULT_TABLE_ACTIVE_PAGE,
  },
  [NetworkDetailsTableType.topNFlowDestination]: {
    ...state.details.queries[NetworkDetailsTableType.topNFlowDestination],
    activePage: DEFAULT_TABLE_ACTIVE_PAGE,
  },
  [NetworkDetailsTableType.tls]: {
    ...state.details.queries[NetworkDetailsTableType.tls],
    activePage: DEFAULT_TABLE_ACTIVE_PAGE,
  },
  [NetworkDetailsTableType.users]: {
    ...state.details.queries[NetworkDetailsTableType.users],
    activePage: DEFAULT_TABLE_ACTIVE_PAGE,
  },
  [NetworkDetailsTableType.http]: {
    ...state.details.queries[NetworkDetailsTableType.http],
    activePage: DEFAULT_TABLE_ACTIVE_PAGE,
  },
});

export const setNetworkQueriesActivePageToZero = (
  state: NetworkModel,
  type: NetworkType
): NetworkQueries | NetworkDetailsQueries => {
  if (type === NetworkType.page) {
    return setNetworkPageQueriesActivePageToZero(state);
  } else if (type === NetworkType.details) {
    return setNetworkDetailsQueriesActivePageToZero(state);
  }
  throw new Error(`NetworkType ${type} is unknown`);
};
