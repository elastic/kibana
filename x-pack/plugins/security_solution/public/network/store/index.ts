/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

import { Reducer, AnyAction } from 'redux';
import * as networkActions from './actions';
import * as networkModel from './model';
import * as networkSelectors from './selectors';
import { NetworkState } from './reducer';

export { networkActions, networkModel, networkSelectors };
export * from './reducer';

export interface NetworkPluginState {
  network: NetworkState;
}

export interface NetworkPluginReducer {
  network: Reducer<networkModel.NetworkModel, AnyAction>;
}
