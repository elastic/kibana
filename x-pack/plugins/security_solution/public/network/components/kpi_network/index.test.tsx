/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { shallow } from 'enzyme';
import React from 'react';
import { Provider as ReduxStoreProvider } from 'react-redux';

import {
  apolloClientObservable,
  mockGlobalState,
  SUB_PLUGINS_REDUCER,
  kibanaObservable,
  createSecuritySolutionStorageMock,
} from '../../../common/mock';
import '../../../common/mock/match_media';
import { createStore, State } from '../../../common/store';
import { NetworkKpiComponent } from '.';

describe('NetworkKpiComponent', () => {
  const state: State = mockGlobalState;
  const props = {
    filterQuery: '',
    from: '2019-06-15T06:00:00.000Z',
    indexNames: [],
    narrowDateRange: jest.fn(),
    setQuery: jest.fn(),
    skip: true,
    to: '2019-06-18T06:00:00.000Z',
  };

  const { storage } = createSecuritySolutionStorageMock();
  let store = createStore(
    state,
    SUB_PLUGINS_REDUCER,
    apolloClientObservable,
    kibanaObservable,
    storage
  );

  beforeEach(() => {
    store = createStore(
      state,
      SUB_PLUGINS_REDUCER,
      apolloClientObservable,
      kibanaObservable,
      storage
    );
  });

  describe('rendering', () => {
    test('it renders the default widget', () => {
      const wrapper = shallow(
        <ReduxStoreProvider store={store}>
          <NetworkKpiComponent {...props} />
        </ReduxStoreProvider>
      );

      expect(wrapper.find('NetworkKpiComponent')).toMatchSnapshot();
    });
  });
});
