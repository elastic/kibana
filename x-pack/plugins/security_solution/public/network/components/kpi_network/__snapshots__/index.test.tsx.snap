// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkKpiComponent rendering it renders the default widget 1`] = `
<NetworkKpiComponent
  filterQuery=""
  from="2019-06-15T06:00:00.000Z"
  indexNames={Array []}
  narrowDateRange={[MockFunction]}
  setQuery={[MockFunction]}
  skip={true}
  to="2019-06-18T06:00:00.000Z"
/>
`;
